app/models/ability.rb

```diff
@@ -6,8 +6,9 @@ class Ability
   def initialize(user)
     # Define abilities for the passed in user here. For example:
     can :manage, Event, user_id: user.id
-    
+    can :destroy, Event, :all if user.owned_events
```

Comment:
You don't need to add new cancan restriction. The one you have in line 8 is good enough. It covers also the case for deletion. You just need to use it everywhere you have delete button for events.

Reply:
Still You don't need to add new cancan restriction.